adagrams/game.py

```diff
@@ -1,11 +1,99 @@
+import random
+
+letter_counts = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+}
+
+
 def draw_letters():
-    pass
+    all_letters = []
+    hand= []
+    for key in letter_counts:
+        for num in range(0, letter_counts[key]):
+            all_letters.append(key)
+    for pepar in range(0, 10):
+        random_index = random.randint(0, len(all_letters)-1)
+        hand.append(all_letters[random_index])
+        del all_letters[(random_index)]
+    return hand
+
+# def draw_letters():
+#     all_letters = []
+#     hand= []
+#     for key in letter_counts:
+#         for num in range(0, letter_counts[key]):
+#             all_letters.append(key)
+#     for pepar in range(0, 10):
+#         random.shuffle(all_letters)
+#         random_index = random.randint(0, len(all_letters)-1)
+#         hand.append(all_letters[random_index])
+#     return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    for char in word:
+        if char not in letter_bank:
+                return False
+        if word.count(char) > letter_bank.count(char):
+            return False
+    return True
+
+
+
 
+
+score_chart = {'A':1, 'E':1, 'I':1, 'O': 1, 'U':1, 'L':1, 'N':1, 'R':1, 'S':1,
+               'T':1, 'D':2, 'G':2, 'B':3, 'C':3, 'M':3, 'P':3, 'F':4, 'H':4,
+               'V':4, 'W':4, 'Y':4, 'k':5, 'J':8, 'X':8, 'Q':10, 'Z':10}
 def score_word(word):
-    pass
+
+    score_list= []
+    for char in word.upper():
+        if char in score_chart.keys():
+            score_list.append(score_chart[char])
+    total_score = sum(score_list)
+    if len(word) >= 7:
+        total_score = total_score + 8
+
+    return total_score
+
+
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    winning_word = ''
+    winning_score = 0
+    for word in word_list:
+        score = score_word(word)
+        if winning_score == score and len(winning_word) != 10:
+            if len(word) == 10 or len(winning_word) > len(word):
+                winning_word = word
+        elif winning_score < score:
+            winning_score = score
+            winning_word = word
```

Comment:
I think I'd prefer lines 96-98 to come before lines 93-95 because the second check is assigning some value to `winning_score` and `winning_word` after they are initialized. 

```python
if winning_score < score:
    winning_score = score
    winning_word = word
elif winning_score == score and len(winning_word) != 10:
    if len(word) == 10 or len(winning_word) > len(word):
        winning_word = word